### DISCUSSION TITLE
about convert .h5 -> onnx

### DISCUSSION THREAD

**Discussion by nadongjin** [leondgarse/Keras_insightface#65] [Q&A]
I have a suggestion for you.
How to create your Keras.h5 model onnx channel first or onnx channel last conversion model.
Personally, I would like to suggest this because onnx is faster and more portable in terms of inference speed.

thanks....

**Comment by leondgarse** [comment]
You can just use [tf2onnx](https://github.com/onnx/tensorflow-onnx). It is still a typical keras model. Basically like:
```py
model = keras.models.load_model("your_own_basic_model.h5")
imgs = np.random.uniform(size=[1, 112, 112, 3]).astype('float32')
preds = model(imgs)

""" Convert """
import tf2onnx
spec = (tf.TensorSpec(model.input_shape, tf.float32, name="input"),)
output_path = model.name + ".onnx"
model_proto, _ = tf2onnx.convert.from_keras(model, input_signature=spec, opset=13, output_path=output_path)

""" Run test """
import onnxruntime as rt
output_names = [n.name for n in model_proto.graph.output]
providers = ['CPUExecutionProvider']
m = rt.InferenceSession(output_path, providers=providers)
onnx_pred = m.run(output_names, {"input": imgs})

print(np.allclose(preds, onnx_pred[0], rtol=1e-5))
```
The last `np.allclose` may return `False` if your model trained using `mixed_float16`, but that doesn't matter.
